### ISSUE TITLE
`MD013` reflow corrupts blockquotes nested in list items (strands `>`, splits admonitions)

### ISSUE THREAD

**Issue by fredrikaverpil** [rvben/rumdl#683]
### rumdl version

0.2.21

### Installation method

Pre-built binary

## Summary

When `MD013` reflow is enabled, reflowing a blockquote that is
**nested inside a list item** corrupts it:

- the blank `>` continuation line loses its `>`, splitting the blockquote into
  two separate blockquotes, and
- when a long line is wrapped, the `>` marker is stranded at the end of the line
  and the wrapped continuation loses its leading `>`.

This breaks
[GitHub-style alerts/admonitions](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts)
(`> [!NOTE]`, etc.) and produces invalid blockquote structure. Top-level
blockquotes (not nested in a list) reflow correctly.

## Notes

- Reproduces with both `reflow-mode = "normalize"` and
  `reflow-mode = "default"`.
- Setting `[MD013] blockquotes = false` does **not** prevent the corruption — it
  only suppresses the length warnings; the reflow fix still rewrites the nested
  blockquote.
- A related issue exists in `MD030`: its list-marker-spacing fix reindents list
  children but drops the `>` on the blank line of a nested blockquote (same kind
  of corruption, triggered without reflow).

### Steps to reproduce

`rumdl.toml`:

```toml
[MD013]
line-length = 80
reflow = true
reflow-mode = "normalize"   # also reproduces with "default"
```

`input.md`:

```markdown
# Title

1. A list item:

   > [!NOTE]
   >
   > A blockquote line inside a list that is long enough that reflow will try to wrap it.

A top-level callout for comparison:

> [!NOTE]
>
> A blockquote line at the top level that is long enough that reflow will try to wrap it.
```

Command:

```sh
rumdl fmt --config rumdl.toml input.md
```

### Expected behavior

The nested blockquote keeps its structure exactly like the top-level one — the
blank line stays `>` and every wrapped line keeps its leading `>` (matching the
indentation):

```markdown
1. A list item:

   > [!NOTE]
   >
   > A blockquote line inside a list that is long enough that reflow will try to
   > wrap it.
```

### Actual behavior

The blank `>` line loses its marker (splitting the blockquote) and the wrapped
continuation line loses its `>`:

```markdown
1. A list item:

   > [!NOTE]

   > A blockquote line inside a list that is long enough that reflow will try to
   wrap it.
```

The top-level blockquote in the same file is reflowed correctly, so the problem
is specific to blockquotes nested in list items.

### Configuration

```toml
[MD013]
line-length = 80
reflow = true
reflow-mode = "normalize"   # also reproduces with "default"
```

### Operating system

macOS

**Comment by fredrikaverpil** [comment]
**Additional example**: this also affects plain (non-admonition) blockquotes.

The corruption isn't specific to `[!NOTE]`-style alerts — *any* blockquote
nested in a list item is affected. A plain `> **bold** …` blockquote spanning
several lines is mangled the same way: the leading `>` markers get pulled into
the prose (stranded mid-line) and the continuation lines lose their `>`, so the
whole blockquote collapses into one paragraph.

Config (same as the original report):

```toml
[MD013]
line-length = 80
reflow = true
reflow-mode = "normalize"
```

`input.md`:

```markdown
# Title

1. Do the setup step:

   > **Why both?** This is a plain blockquote that is not a `[!NOTE]` admonition.
   > It spans several source lines, and reflow strands the `>` marker at the end
   > of each line when it rewraps them together.
```

`rumdl fmt --config rumdl.toml input.md` produces (note the stray `>` characters
mid-paragraph and the lost blockquote markers):

```markdown
# Title

1. Do the setup step:

   > **Why both?** This is a plain blockquote that is not a `[!NOTE]`
   admonition. > It spans several source lines, and reflow strands the `>`
   marker at the end > of each line when it rewraps them together.
```

**Comment by rvben** [comment]
Thanks @fredrikaverpil. Fixed in **v0.2.22**.

MD013 reflow was collecting an indented blockquote inside a list item as plain prose, so the `>` markers got rewrapped as words and continuations lost their `>`. List-item reflow now hands a nested blockquote to the blockquote-aware path, which preserves the `>` prefix (including the list indent). Covers both examples here, admonition or plain.
